Buildings/Fluid/HeatExchangers/DXCoils/AirSource/Examples/SingleSpeedHeating.mo

```diff
@@ -0,0 +1,158 @@
+within Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Examples;
+model SingleSpeedHeating "Test model for single speed DX heating coil"
+  extends Modelica.Icons.Example;
+
+  package Medium = Buildings.Media.Air
+    "Fluid medium for the model";
+
+  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal=datCoi.sta[datCoi.nSta].nomVal.m_flow_nominal
+    "Nominal mass flow rate";
+
+  parameter Modelica.Units.SI.PressureDifference dp_nominal=1000
+    "Pressure drop at m_flow_nominal";
+
+  parameter
+    Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.BaseClasses.CoilHeatTransfer
+    datCoi(
+    is_CooCoi=false,
+    nSta=1,
+    minSpeRat=0.2,
+    sta={
+      Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.BaseClasses.Stage(
+      spe=1800/60,
+      nomVal=
+        Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.BaseClasses.NominalValues(
+        Q_flow_nominal=16381.47714,
+        COP_nominal=3.90494,
+        SHR_nominal=1,
+        m_flow_nominal=2,
+        TEvaIn_nominal=273.15 - 5,
+        TConIn_nominal=273.15 + 21),
+    perCur=
+      Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Examples.PerformanceCurves.DXHeating_Curve_II())})
+    "Coil heat transfer data record"
+    annotation (Placement(transformation(extent={{60,34},{80,54}})));
+
+  parameter
+    Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.BaseClasses.Defrost
+    datDef(
+    final defOpe=Buildings.Fluid.HeatExchangers.DXCoils.BaseClasses.Types.DefrostOperation.resistive,
+    final defTri=Buildings.Fluid.HeatExchangers.DXCoils.BaseClasses.Types.DefrostTimeMethods.timed,
+    final tDefRun=1/6,
+    final TDefLim=273.65,
+    final QDefResCap=10500,
+    final QCraCap=200)
+    "Heating coil defrost data"
+    annotation (Placement(transformation(extent={{60,0},{80,20}})));
+
+
+  Buildings.Fluid.HeatExchangers.DXCoils.AirSource.SingleSpeedHeating sinSpeDX(
+    datCoi(
+      final nSta=datCoi.nSta,
+      final minSpeRat=datCoi.minSpeRat,
+      final sta=datCoi.sta),
+    redeclare package Medium = Medium,
+    final dp_nominal=dp_nominal,
+    final T_start=datCoi.sta[1].nomVal.TConIn_nominal,
+    final show_T=true,
+    final from_dp=true,
+    final energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
+    final datDef=datDef)
```

Comment:
This design needs to be changed: `Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.HeatingCoil` has already everything of `datCoi` and it has `datRef`. Why does this example instantiated `datCoi` and `datRef` and then assigns both to the coil model?
You should be able to just state
```
... sinSpeDX(
  datCoi = datCoi,
....
```